src/core/derivation.ts

```diff
@@ -185,4 +197,58 @@ export function untrackedStart() {
 
 export function untrackedEnd(prev: boolean) {
 	globalState.isTracking = prev;
-}
\ No newline at end of file
+}
+
+export function changeDependenciesState(targetState, derivation: IDerivation) {
+	const oldState = derivation.dependenciesState;
+	if (oldState === targetState) return;
+	derivation.dependenciesState = targetState;
+	const obs = derivation.observing;
+	const l = obs.length;
+
+	const m = derivation.__mapid;
+
+	// just because it can be bottleneck
+	if (oldState === 0) {
```

Comment:
I'm right now trying to change whole `DerivationsSets` to something more performant, because it seems to be a performance bottleneck.